apps/aechannel/src/aesc_fsm.erl

```diff
@@ -936,55 +940,49 @@ awaiting_signature(cast, {?SIGNED, withdraw_tx, Tx} = Msg,
 awaiting_signature(cast, {?SIGNED, ?FND_CREATED, SignedTx} = Msg,
                    #data{role = responder,
                          latest = {sign, ?FND_CREATED, HSCTx, Updates}} = D) ->
-    NewSignedTx = aetx_sign:add_signatures(
-                    HSCTx, aetx_sign:signatures(SignedTx)),
```

Comment:
Who does now check that `SignedTx` is related to `HSCTx` rather than being a completely distinct transaction?

Reply:
Is the assumption maybe that the user of the user API is trusted?

Reply:
Assumption is that if it is a different:
* new transaction - then authentication method will not match
* old co-signed transaction - the check round will catch it